build-scripts/update_sds_to_scap_1_3.py

```diff
@@ -0,0 +1,139 @@
+        # The component ID is the component-ref href without leading '#'
+        component_id = component_ref_href[1:]
+
+        # Locate the <xccdf:check> element of an <xccdf:Rule> with id security_patches_up_to_date
+        component = None
+        oval_check = None
+        components = datastreamtree.findall(".//{%s}component" % ds_ns)
+        for component in components:
+            if component.get('id') == component_id:
+                component = component
+        if component is None:
+            # Something strange happened
+            sys.stderr.write("Couldn't find <component> %s referenced by <component-ref> %s" %
+                             (component_id, component_ref_id))
+            sys.exit(1)
+
+        rules = component.findall(".//{%s}Rule" % (xccdf_ns))
+        for rule in rules:
+            if rule.get('id').endswith('rule_security_patches_up_to_date'):
+                rule_checks = rule.findall("{%s}check" % xccdf_ns)
+                for check in rule_checks:
+                    if check.get('system') == oval_ns:
+                        oval_check = check
+                        break
+
+        if oval_check is None:
+            # The component doesn't have a security patches up to date rule with an OVAL check
+            continue
+
+        # SCAP 1.3 demands multi-check true if the Rules security_patches_up_to_date is
+        # evaluated by multiple OVAL patch class definitinos.
+        # See 3.2.4.3, SCAP 1.3 standard (NIST.SP.800-126r3)
+        oval_check.set('multi-check', 'true')
+
+        check_content_ref = oval_check.find('{%s}check-content-ref' % xccdf_ns)
+        href_url = check_content_ref.get('href')
+
+        # Use URL's path to define the component name and URI
+        # Path attribute returned from urlparse contains a leading '/', when mangling it
+        # it will get replaced by '-'.
+        # Let's strip the '/' to avoid a sequence of "_-" in the component-ref ID.
+        component_ref_name = mangle_path(urlparse(href_url).path[1:])
+        component_ref_uri = component_ref_prefix + component_ref_name
+
+        # update @href to refer the datastream component name
+        check_content_ref.set('href', component_ref_name)
+
+        # Add a uri refering the component in Rule's Benchmark component-ref catalog
+        uri_exists = False
+        catalog = component_ref.find('{%s}catalog' % cat_ns)
+        uris = catalog.findall("{%s}uri" % (cat_ns))
+        for uri in uris:
+            if uri.get('name') == component_ref_name:
+                uri_exists = True
+                break
+        if not uri_exists:
+            uri = ssg.xml.ElementTree.Element('{%s}uri' % cat_ns)
+            uri.set('name', component_ref_name)
+            uri.set('uri', component_ref_uri)
+            catalog.append(uri)
+
+        # The component-ref ID is the catalog uri without leading '#'
+        component_ref_feed_id = component_ref_uri[1:]
+
```

Comment:
When I build RHEL 7 data stream I get in scap_org.open-scap_cref_ssg-rhel7-pcidss-xccdf-1.2.xml component-ref this element:
`<ns2:uri name="ecurity-data-oval-com.redhat.rhsa-RHEL7.xml.bz2" uri="#scap_org.open-scap_cref_ecurity-data-oval-com.redhat.rhsa-RHEL7.xml.bz2"/>`. There is missing `s` 2 times.

Reply:
Nice catch, I ended up introducing a bug during manual filtering.

Reply:
Commits 15c70d0 and 304270f rename variables to make them clear about what components they are about.